From 9c7dde3d591aafa4b83191a9cddf3e59812ec95f Mon Sep 17 00:00:00 2001
From: Jean Trivelly <jean.trivelly@intel.com>
Date: Tue, 19 Mar 2013 18:14:22 +0100
Subject: [PORT FROM MAIN] wifi: Add a property to select SoftAP channel

BZ: 94270

Add a property to be able to choose the channel
to be used by SoftAP.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: yes

Change-Id: Ib57e69cf75c52e7a92815cb1cacf8ad674baa141
Orig-Change-Id: I27d3c3267fd4fc22a1c53a6ea3d62b0a75c6403f
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 SoftapController.cpp |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/SoftapController.cpp b/SoftapController.cpp
index 471f09d..60eb6ab 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -35,6 +35,7 @@
 
 #define LOG_TAG "SoftapController"
 #include <cutils/log.h>
+#include <cutils/properties.h>
 #include <netutils/ifc.h>
 #include <private/android_filesystem_config.h>
 #include "wifi.h"
@@ -158,11 +159,16 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
 
     char *wbuf = NULL;
     char *fbuf = NULL;
+    char ap_channel_s[10];
+    int ap_channel = 6;
+    
+    property_get("wifi.ap.channel", ap_channel_s, "6");
+    ap_channel = atoi(ap_channel_s);
 
     asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
-            "/data/misc/wifi/hostapd\nssid=%s\nchannel=6\nieee80211n=1\n"
+            "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
             "hw_mode=g\n",
-            argv[2], argv[3]);
+	     argv[2], argv[3], ap_channel);
 
     if (argc > 4) {
         if (!strcmp(argv[4], "wpa-psk")) {
-- 
1.7.4.1

